Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide.hs

```diff
@@ -426,7 +426,7 @@ recalculateFareForDistance ServiceHandle {..} booking ride recalcDistance thresh
           actualDistance = Just recalcDistance,
           estimatedDistance = Just oldDistance,
           rideTime = booking.startTime,
-          waitingTime = secondsToMinutes . roundToIntegral <$> (diffUTCTime <$> ride.tripStartTime <*> ride.driverArrivalTime),
+          waitingTime = if isNothing ride.driverArrivalTime then Nothing else fmap (max 0) (secondsToMinutes . roundToIntegral <$> (diffUTCTime <$> ride.tripStartTime <*> (liftA2 max ride.driverArrivalTime (Just booking.startTime)))),
```

Comment:
What is this logic?

Reply:
This is for scheduled rides, we should take `tripStartTime - max(startTime, driverArrivalTime)` as waiting time